src/lib.rs

```diff
@@ -259,6 +261,29 @@ mod contract {
         result.map(|_f| Vec::new()).sdk_process();
     }
 
+    #[no_mangle]
+    pub extern "C" fn register_relayer() {
+        let relayer_address = sdk::read_input();
+        let mut engine = Engine::new(predecessor_address());
+        engine.register_relayer(
+            sdk::predecessor_account_id().as_slice(),
+            Address(relayer_address.as_slice().try_into().unwrap()),
+        );
+    }
+
+    /// Allow receiving NEP141 tokens to the EVM contract
+    #[no_mangle]
+    pub extern "C" fn ft_on_transfer() {
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        let current_account_id = String::from_utf8(sdk::current_account_id()).unwrap();
```

Comment:
If these are 100% infallible, you can instead use `from_utf8_unchecked` as it is coming from the sdk. It'll reduce the cost of the tx.

Reply:
I skip this with a suggestion from @birchmd

Reply:
I think we should do this inside `sdk::*_account_id()` given that we don't want to use unsafe for now here